inbm/cloudadapter-agent/cloudadapter/cloud/client/connections/mqtt_connection.py

```diff
@@ -95,8 +95,10 @@ def _set_proxy(self, config: ProxyConfig) -> None:
         @param config: (ProxyConfig) Config to use
         """
         if config.endpoint:
-            socks.set_default_proxy(socks.PROXY_TYPE_HTTP, *config.endpoint)
-            socket.socket = socks.socksocket  # type: ignore
+            proxy_addr, proxy_port = config.endpoint
+            self._client.proxy_set(proxy_type=socks.HTTP,
+                                   proxy_addr=proxy_addr,
+                                   proxy_port=proxy_port)
```

Comment:
**<h2>$\color{red}Defects: ErrorHandling$</h2>**
**Suggestion:** Validate the proxy_port to ensure it is within the acceptable range before setting the proxy configuration on the MQTT client.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if config.endpoint:
            proxy_addr, proxy_port = config.endpoint
            if not 0 <= proxy_port <= 65535:
                raise ValueError(f"Proxy port {proxy_port} is out of the valid range (0-65535).")
            self._client.proxy_set(proxy_type=socks.HTTP,
                                   proxy_addr=proxy_addr,
                                   proxy_port=proxy_port)
```